tests/console/validate_partition_table.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary:  Validate partition table via program 'parted' or 'blkid'.
+#
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+
+use scheduler 'get_test_suite_data';
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
+    my $disks = get_test_suite_data()->{disks};
```

Comment:
it is better to avoid too many assignation (for maintenability), you can use this inside the foreach.